cognite/client/data_classes/filters.py

```diff
@@ -319,6 +400,23 @@ def _filter_body(self, camel_case_property: bool) -> list:
 
 @final
 class Range(FilterWithProperty):
+    """Filters results based on a range of values.
+
+    Args:
+        property (PropertyReference): The property to filter on.
+        gt (FilterValue | None): Greater than.
+        gte (FilterValue | None): Greater than or equal to.
+        lt (FilterValue | None): Less than.
+        lte (FilterValue | None): Less than or equal to.
```

Comment:
Just noticed that these type hints using `FilterValue` are wrong/misleading as only a subset is supported (other misleading uses: `Overlaps` and `Prefix`):

> (RangeValue (RangeValue (string) or RangeValue (number) or RangeValue (integer))) or ReferencedPropertyValueV3 (object) (FilterValueRange)

```py
@dataclass
class PropertyReferenceValue:
    property: PropertyReference

@dataclass
class ParameterValue:
    parameter: str

FilterValue: TypeAlias = Union[RawValue, PropertyReferenceValue, ParameterValue]
RawValue: TypeAlias = Union[str, float, bool, Sequence, Mapping[str, Any], Label]
```

Reply:
Yeah, not ideal. Will leave some validation for runtime. But not gonna address that in this PR.